mantidproject/mantid
# SaveAscii: add possibility to save the values of the spectrum…

**Description of work.**
It was requested that SaveAscii write also the spectrum axis value.
Although it already offers something like that (the input called `SpectrumMetaData`), it is limited to few observables like Q, etc.
This PR adds a new flag (not to intervene with the one above), that makes it write the spectrum axis values no matter the axis type. For the `BinEdgeAxis` it computes on-the-fly the bin centre value.

**To test:**
Load the attached file to `sqw`. First cut off the `.zip`; it is a nexus file.

[sqw.nxs.zip](https://github.com/mantidproject/mantid/files/3652547/sqw.nxs.zip)

```
SaveAscii(InputWorkspace='sqw', WriteSpectrumAxisValue=True, Filename='~/Desktop/test.dat')
```
Open the test.dat in the editor, you should see at line 2 the same value as on the spectrum (a.k.a. vertical) axis of the workspace in mantid matrix view.



Fixes #26862  




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
